lib/github/kv.rb

```diff
@@ -252,6 +253,80 @@ def setnx(key, value, expires: nil)
       }
     end
 
+    # increment :: String, Integer, expires: Time? -> Integer
+    #
+    # Increment the key's value by an amount.
+    #
+    # key     - The key to increment.
+    # amount  - The amount to increment the key's value by.
+    # expires - When the key should expire.
+    #
+    # Returns the key's value after incrementing.
+    def increment(key, amount: 1, expires: nil)
+      raise ArgumentError.new("The amount specified must be an integer") unless 1.is_a? Integer
+      raise ArgumentError.new("The amount specified my be > 0") if amount < 1
+
+      # This query uses a few MySQL "hacks" to ensure that the incrementing
+      # is done atomically and the value is returned. The first trick is done
+      # using the `LAST_INSERT_ID` function. This allows us to manually set
+      # the LAST_INSERT_ID returned by the query. Here we are able to set it
+      # to the new value when an increment takes place, essentally allowing us
+      # to do: `UPDATE...;SELECT value from key_value where key=:key` in a
+      # single step.
+      #
+      # However the `LAST_INSERT_ID` trick is only used when the value is
+      # updated. Upon a fresh insert we know the amount is going to be set
+      # to the amount specified.
+      #
+      # Lastly we only do these tricks when the value at the key is and integer.
+      # If the value is not an integer the update ensures the values remain the
+      # same and we raise an error.
+      sql = GitHub::SQL.run(<<-SQL, key: key, amount: amount, now: now, expires: expires || GitHub::SQL::NULL, connection: connection)
```

Comment:
This is a really long line already. Let's do `expires ||= GitHub::SQL::NULL` on a previous line instead of inlining it here. Alternatively, you could make it the default value for the `expires:` kwarg.